### PR TITLE
Ensure a completion message is present after setting up totp.

### LINKED ISSUE #302: First time login improved flow
It'd be best from UX point of view to add a confirmation screen before the automatic redirect to the application. So the flow should be something like this:
image
Or in screens it would be 

1. Trying to sign in for the first time 
image

2. Required TOTP setup
image

3. Setup complete, message to confirm that action has been successful (also good place to show any errors if some occur). And automatic redirect within 3 seconds.
image

4. Landing on the 3rd party application page
image

_Originally posted by @lukasSerelis in https://github.com/canonical/identity-platform-login-ui/issues/275#issuecomment-2376117255_

### PR DESCRIPTION
**Pull request by edlerd**
# Done
- Ensure a completion message is present after setting up totp
- Automatically redirect back to the application
- Fix wrong page URL replacement, when arriving at the setup secure page.

fixes #302

### CODE REVIEW

File: ui/pages/setup_secure.tsx [hunk 1]
```diff
@@ -81,11 +81,11 @@ const SetupSecure: NextPage = () => {
         })
         .then(({ data }) => {
           if ("redirect_to" in data) {
-            window.location.href = data.redirect_to as string;
+            window.location.href = `./setup_complete?flow=${data.id}`;
             return;
           }
```
Comment (nsklikas) [inline ui/pages/setup_secure.tsx:86]: This is not needed, we can remove this case. The backend should always return the flow

**Review by nsklikas — COMMENTED** [review]
LGTM, only a small comment

File: ui/pages/setup_secure.tsx (hunk 1 above)
Comment (edlerd) [reply ui/pages/setup_secure.tsx]: removed this bit.

File: ui/pages/setup_secure.tsx [hunk 2]
```diff
@@ -80,12 +80,11 @@ const SetupSecure: NextPage = () => {
...
-          }
           if (data?.return_to) {
-            window.location.href = data?.return_to;
+            const flowParam = data.return_to.endsWith("/setup_complete")
+              ? ""
+              : `?flow=${data.id}`;
+            window.location.href = `./setup_complete${flowParam}`;
             return;
           }
           window.location.href = "./error";
```
Comment (nsklikas) [inline ui/pages/setup_secure.tsx:90]: I would remove this condition, let the other page handle if there is no return_to. If you want to be safe, we can check the state of the flow (but I'm not sure if that's needed)

IMO Redirecting to the error page does not make sense

File: ui/pages/setup_secure.tsx [hunk 3]
```diff
@@ -80,12 +80,11 @@ const SetupSecure: NextPage = () => {
...
-          if ("redirect_to" in data) {
-            window.location.href = data.redirect_to as string;
-            return;
-          }
           if (data?.return_to) {
-            window.location.href = data?.return_to;
+            const flowParam = data.return_to.endsWith("/setup_complete")
+              ? ""
+              : `?flow=${data.id}`;
+            window.location.href = `./setup_complete${flowParam}`;
```
Comment (nsklikas) [inline ui/pages/setup_secure.tsx:87]: What if the return_to page is not `setup_complete` but something else? Shouldn't this be:
```suggestion
            window.location.href = `./${data.return_to}${flowParam}`;
```
?

File: ui/pages/setup_secure.tsx (hunk 2 above)
Comment (nsklikas) [inline ui/pages/setup_secure.tsx:90]: I don't think we should redirect to the error page in this case, let's just send them to the `setup_complete` page with the flow and let that page handle it.

**Review by nsklikas — APPROVED** [review]

**Review by nsklikas — APPROVED** [review]

File: ui/pages/setup_secure.tsx (hunk 3 above)
Comment (nsklikas) [reply ui/pages/setup_secure.tsx]: you are right

**Review by shipperizer — APPROVED** [review]

**Pull request merged**